src/ape_pie/client.py

```diff
@@ -85,16 +82,14 @@ def __init__(
 
         self.base_url = base_url
 
-        # set the attributes that requests.Session supports directly, but only if an
-        # actual value was provided.
-        for attr in self.__attrs__:
-            val = request_kwargs.pop(attr, sentinel)
-            if val is sentinel:
-                continue
-            setattr(self, attr, val)
-
-        # store the remainder so we can inject it in the ``request`` method.
-        self._request_kwargs = request_kwargs
+        self._request_kwargs = {}
+        # set the attributes that requests.Session supports directly
+        for k, v in request_kwargs.items():
+            if k in self.__attrs__:
+                setattr(self, k, v)
+            else:
+                # store the remainder so we can inject it in the ``request`` method.
+                self._request_kwargs[k] = v
```

Comment:
I chose not to annotate `self._request_kwargs` as `RequestKwargs`, because mypy only allows assignment in it with literals. The fact that `k` came from the same type...

PyRight either is smarter or doesn't seem to mind.

Reply:
I'm okay with switching to pyright in CI instead of mypy if that makes life better